### PR TITLE
Export all Interfaces used as Parameters added to Style Guide [Probably Controversial]

### PR DESCRIPTION
**Pull request by CodingCarter**
I think that we should have to export all of the interfaces that are used as parameters inside of exported functions and classes in our modules because if someone creates a variable that stores the data that is for the interface, it is better to set the interfaces of that object instead of just have it as a generic object.  Here is an example:

```ts
// BAD - only function exported
import { createPerson } from "import_link";

const personData = {
  name: "Carter",
  age: 12,
  isCoding: true
}

console.log(createPerson(personData));

// GOOD - only function and interface that is used as parameter exported
import { Person, createPerson } from "import_link";

const personData: Person = {
  name: "Carter",
  age: 12,
  isCoding: true
}

console.log(createPerson(personData));
```

This will also be better for debugging because the people can actually have the interface to use in their own test code.  Anyways, thanks for reading and I hope you consider adding this in!

### CODE REVIEW

**Comment by nayeemrmn** [comment]
Do we need to include general TS best practices in the style guide? I guess you could argue "Use JSDoc" is there, but in that case we use it to communicate project-specific preferences about what and what not to document.

**Comment by luke-john** [comment]
Just happened to come across this from the discord server. 

An alternative for the consumer to have typesafety without exporting the parameters type is possible using the [Parameters generic](https://www.typescriptlang.org/docs/handbook/utility-types.html#parameterstype) as follows.

```ts
// my_file.ts
interface Person {
  name: string;
  age: number;
}
export function createPerson(person: Person) {
  return { name, age };
}
// mod.ts
export { createPerson } from "./my_file.ts";

// Alternative - only function and interface that is used as parameter exported
import { createPerson } from "import_link";

const personData: Parameters<typeof createPerson>[0] = {
  name: "Carter",
  age: 12,
  isCoding: true
}

console.log(createPerson(personData));
```
(note: the my_file in this example is modified from the one in the diff, as you seem to not be using the Person interface as an argument, given the description this appears to be a mistake)

File: docs/contributing/style_guide.md [hunk 1]
```diff
@@ -174,6 +174,25 @@ export interface PWrite {
...
+  name: string;
+  age: number;
+}
+
+export function createPerson(name: string, age: number): Person {
+  return { name, age };
+}
+
+// mod.ts
+export { Person, createPerson } from "./my_file.ts";
```
Comment (kitsonk) [inline docs/contributing/style_guide.md:193]: In order to support `--no-check` this needs to be:

```ts
export { createPerson } from "./my_file.ts";
export type { Person } from "./my_file.ts";
```

**Comment by crowlKats** [comment]
"must" is the wrong word here; "should" would be more appropriate.

File: docs/contributing/style_guide.md (hunk 1 above)
Comment (kitsonk) [reply docs/contributing/style_guide.md:193]: nit, also should be alphabetical

**Comment by CodingCarter** [comment]
> "must" is the wrong word here; "should" would be more appropriate.

Isn't a style guide supposed to enforce a way of coding to standardize the design?  I think we would just be better off not adding this if we are putting "should".

File: docs/contributing/style_guide.md (hunk 1 above)
Comment (CodingCarter) [reply docs/contributing/style_guide.md]: > nit, also should be alphabetical

Sorry if I am just being slow, but what is nit?  Do you mean the interface labels?

**Comment by crowlKats** [comment]
@CodingCarter there is no way to enforce this, but only to have the community follow the "convention". "must" makes it sound like it has to be that way or else things will not work.

File: docs/contributing/style_guide.md (hunk 1 above)
Comment (caspervonb) [reply docs/contributing/style_guide.md]: Nit as in "nitpick".

**Comment by CodingCarter** [comment]
> @CodingCarter there is no way to enforce this, but only to have the community follow the "convention". "must" makes it sound like it has to be that way or else things will not work.

Ahh...I understand that.  I agree with your thinking I'll change it.

File: docs/contributing/style_guide.md (hunk 1 above)
Comment (CodingCarter) [reply docs/contributing/style_guide.md]: > nit, also should be alphabetical

Thanks @caspervonb for clearing what "nit" means.  I usually order my imports in the order that I use them, but I will start using your convention👍

Comment (kitsonk) [reply docs/contributing/style_guide.md]: [Yes, sorry](https://www.urbandictionary.com/define.php?term=nit)

**Review by kitsonk — CHANGES_REQUESTED** [review]
You need to run linting on this to get the CI to go green.

**Review by kitsonk — APPROVED** [review]

**Pull request merged**